### ISSUE TITLE
'Unnecessary type casting' false positive

### ISSUE THREAD

**Issue by 3DFace** [kalessil/phpinspectionsea#492]
Code to type:

```php
function multiply_by_100_and_truncate(float $x) {
	return (int)($x * 100); // false positive here, type casting makes job 
}

echo multiply_by_100_and_truncate(10.021);
```

At Version: 2.3.10-preview

**Comment by kalessil** [comment]
Duplicated #490. Please vote for the JB issue linked there.

You can already take the binary from the master branch where this is fixed already.
